OrbitCore/TracepointEventBuffer.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CORE_TRACEPOINT_EVENT_BUFFER_H_
+#define ORBIT_CORE_TRACEPOINT_EVENT_BUFFER_H_
+
+#include <set>
+
+#include "BlockChain.h"
+#include "SamplingProfiler.h"
+#include "Threading.h"
+#include "capture_data.pb.h"
+
+class TracepointEventBuffer {
+ public:
+  TracepointEventBuffer() : m_MaxTime(0), m_MinTime(LLONG_MAX) {}
+
+  void Reset() {
+    tracepoint_events_.clear();
+    m_MinTime = LLONG_MAX;
+    m_MaxTime = 0;
+  }
+  std::map<ThreadID, std::map<uint64_t, orbit_client_protos::TracepointEventInfo> >&
+  GetTracepoints() {
+    return tracepoint_events_;
+  }
+  Mutex& GetMutex() { return m_Mutex; }
```

Comment:
Exporting the mutex seems a bit weird. Wild guess: Someone uses GetTracepoints and GetMutex to get access to the private data of the class and modifies it from the outside. Consider providing member functions for this instead (well, if I am right about that...).

Reply:
The data from TracepiintEventBuffer is read in another function and, in order for the it to not be corrupted that mutex is used.